### PR TITLE
Finetune Say It With Flowers quest

### PR DESCRIPTION
**Pull request by ymkc**

### CODE REVIEW

File: scripts/zones/Windurst_Waters/npcs/Ohbiru-Dohbiru.lua [hunk 1]
```diff
@@ -221,38 +221,66 @@ function onEventFinish(player,csid,option)
...
+            elseif (option == 4) then
+                if (player:getGil() >= 200) then
+                    player:addItem(957); -- Amaryllis
+                    player:delGil(200);
+                    player:messageSpecial(ITEM_OBTAINED,957);
+                    player:needToZone(true);
+                else
+                    player:messageSpecial(NOT_HAVE_ENOUGH_GIL);
+                end
+            elseif (option == 5) then
```
Comment (takhlaq) [inline scripts/zones/Windurst_Waters/npcs/Ohbiru-Dohbiru.lua]: not sure we want so many lines of repeated code
this whole tree can be put in a table
```lua
local tabre = 
{
    -- [key] = { itemid = id, gil = val },
    [5] = { itemid = 958, gil = 100 },
    ...
}
local choice = tabre[option]
if choice and player:getGil() >= choice.gil then
    player:addItem(choice.item)
    ...
end
```

File: scripts/zones/Windurst_Waters/npcs/Moari-Kaaori.lua [hunk 2]
```diff
@@ -73,36 +73,41 @@ function onEventFinish(player,csid,option)
...
+        player:needToZone(true);
         player:setVar("FLOWER_PROGRESS",0);
     elseif (csid == 525) then -- Repeatable quest rewards.
         player:tradeComplete();
         player:addFame(WINDURST,30);
-        player:addGil(GIL_RATE*400);
+        player:addGil(GIL_RATE * 400);
         player:setVar("FLOWER_PROGRESS",0);
-        player:setLocalVar("FLOWER_ZONE",1);
+        player:needToZone(true);
```
Comment (takhlaq) [inline scripts/zones/Windurst_Waters/npcs/Moari-Kaaori.lua]: can also table this

Comment (ymkc) [reply scripts/zones/Windurst_Waters/npcs/Moari-Kaaori.lua]: hmm, not quite sure what to table on

Comment (takhlaq) [reply scripts/zones/Windurst_Waters/npcs/Moari-Kaaori.lua]: ah i misread the code above, should be fine

**Pull request merged**